### PR TITLE
CDAP-3563 Backporting the fix from 3.2 to 3.0

### PR DESCRIPTION
**Pull request by sagarkapare**
JIRA: https://issues.cask.co/browse/CDAP-3563
Back porting PR #3955 and some cleanup to 3.0.6

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRuntimeService.java [hunk 1]
```diff
@@ -391,28 +379,12 @@ public ProgramLiveInfo getLiveInfo(Id.Program program, ProgramType type) {
...
-    private final PathFilter namespacedFilter;
     private final List<URL> rmUrls;
-    private final String namespace;
-    private FileSystem hdfs;
 
-    public ClusterResourceReporter(MetricsCollectionService metricsCollectionService, Configuration hConf,
-                                   CConfiguration cConf) {
+    public ClusterResourceReporter(MetricsCollectionService metricsCollectionService, Configuration hConf) {
       super(metricsCollectionService.getCollector(
         ImmutableMap.<String, String>of()));
```
Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRuntimeService.java]: fix the indentation here?

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRuntimeService.java [hunk 2]
```diff
@@ -391,28 +379,12 @@ public ProgramLiveInfo getLiveInfo(Id.Program program, ProgramType type) {
    */
   private class ClusterResourceReporter extends AbstractResourceReporter {
     private static final String RM_CLUSTER_METRICS_PATH = "/ws/v1/cluster/metrics";
-    private final Path hbasePath;
```
Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRuntimeService.java]: if this stuff is unused, we should fix it on develop as well. obviously not this PR.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRuntimeService.java [hunk 3]
```diff
@@ -391,28 +379,12 @@ public ProgramLiveInfo getLiveInfo(Id.Program program, ProgramType type) {
...
-    private final Path namedspacedPath;
-    private final PathFilter namespacedFilter;
     private final List<URL> rmUrls;
-    private final String namespace;
-    private FileSystem hdfs;
 
-    public ClusterResourceReporter(MetricsCollectionService metricsCollectionService, Configuration hConf,
-                                   CConfiguration cConf) {
+    public ClusterResourceReporter(MetricsCollectionService metricsCollectionService, Configuration hConf) {
```
Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRuntimeService.java]: can you remove the public modifier? its inside a private class, so is redundant

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRuntimeService.java (hunk 2 above)
Comment (bdmogal) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRuntimeService.java]: nvm. just checked that this has been removed from develop already

**Comment by bdmogal** [comment]
just a couple comments. the rest LGTM

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRuntimeService.java (hunk 1 above)
Comment (sagarkapare) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRuntimeService.java]: fixed.

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRuntimeService.java (hunk 3 above)
Comment (sagarkapare) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/DistributedProgramRuntimeService.java]: removed.

**Comment by sagarkapare** [comment]
Addressed review comments. Release build is running here - http://builds.cask.co/browse/CDAP-RB308-1

**Comment by sagarkapare** [comment]
Build passed. Merging.

**Pull request merged**